### PR TITLE
feat: improve types, close #970

### LINKED ISSUE #970: Issues with typings in 2.1.0
I've been using this under Typescript, and I've found some issues with version 2.1.0.

1. Viewer does not have a "setValue" method. In version 1.4.10 it did, but it just called "setMarkdown".
2. If you're adding custom toolbar buttons, you need to add event types for your buttons to the eventManager. The examples do this as "editor.eventManager.addEventType('clickCustomButton');", but "eventManager" is not available in the index.d.ts. Is this the intended way of doing it?
3. Editor's "previewStyle" could be set to 'none' (or something other than the two allowed values) to not have a preview in the markdown. Null still works, and might be the intended way of doing things.

Would you accept a pull request to fix any of these, and which if yes?

### PR DESCRIPTION
**Pull request by midgleyc**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

As discussed in #970, I've found some types that I'd like to change. I've left each separate change in its own commit for easier discussion / reversion if necessary. I've tried to mark whether it was a fix (type was wrong) or feat (new public type).

setValue was present under the functions of Viewer, but calling it crashed the application as it doesn't exist any more.

When using custom events, you can pass additional arguments through emit. The first arg is always the name of the event, so I left it in.

eventManager wasn't available under Editor: I added a method to call-through for the method you need to call when adding a custom button, and updated the example to use it. I also noted that addCommand should be public: there is the same issue with commandManager.

CommandManagerExecFunc incorrectly said the first argument was a string -- I think it's actually one of Editor (for global commands), MarkdownEditor (for markdown commands) or WysiwygEditor (for wysiwyg commands).

Finally, there are some methods in MarkdownEditor that are accessible through CodeMirrorExt (that MarkdownEditor extends) that are useful when writing custom commands. I added the two I used.

Closes #970

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/editor/index.d.ts [hunk 1]
```diff
@@ -570,8 +570,19 @@ declare namespace toastui {
     public styleToSelectedCells(onStyle: SquireExt, options?: object): void;
   }
 
+  interface MarkdownRange {
+    from: CodeMirror.Position;
+    to: CodeMirror.Position;
+    ollapsed: boolean;
```
Comment (seonim-ryu) [inline apps/editor/index.d.ts:576]: The value of this property is wrong, `collapsed ` is correct.

File: apps/editor/index.d.ts [hunk 2]
```diff
@@ -729,6 +740,12 @@ declare namespace toastui {
 
     public addWidget(selection: Range, node: Node, style: string, offset?: number): void;
 
+    public addCommand(type: Command): void;
+
+    public addCommand(type: string, props: CommandPropsOptions): void;
```
Comment (seonim-ryu) [inline apps/editor/index.d.ts:745]: Why did you declare `addCommand twice` ?

File: apps/editor/index.d.ts [hunk 3]
```diff
@@ -835,8 +852,6 @@ declare namespace toastui {
 
     public setMarkdown(markdown: string): void;
 
-    public setValue(markdown: string): void;
```
Comment (seonim-ryu) [inline apps/editor/index.d.ts:838]: 👍 
It remains in the `Editor` class, can you remove it too?
> https://github.com/nhn/tui.editor/blob/7cddc9eca1feb232aa6be06c07ee855ea8759742/apps/editor/index.d.ts#L802

File: apps/editor/index.d.ts [hunk 4]
```diff
@@ -570,8 +570,19 @@ declare namespace toastui {
...
+  }
+
+  class CodeMirrorExt {
+    getCurrentRange(): MarkdownRange;
+    getEditor(): CodeMirrorType;
+  }
+
   // @TODO: change toastMark type definition to @toast-ui/toastmark type file through importing
-  class MarkdownEditor {
+  class MarkdownEditor extends CodeMirrorExt {
```
Comment (seonim-ryu) [inline apps/editor/index.d.ts:585]: Why did you add the `CodeMirrorExt` type to handle the event issue?

File: apps/editor/index.d.ts (hunk 2 above)
Comment (midgleyc) [reply apps/editor/index.d.ts:745]: The first type is different: the first one takes a single Command, the second one takes two parameters: a string and a CommandPropsOptions

https://github.com/nhn/tui.editor/blob/master/apps/editor/src/js/editor.js#L359-L365

I could declare it 

> public addCommand(type: string | Command, props?: CommandPropsOptions): void;

but I thought declaring the two options separately was more precise.

File: apps/editor/index.d.ts (hunk 1 above)
Comment (midgleyc) [reply apps/editor/index.d.ts:576]: good point, I will fix this

File: apps/editor/index.d.ts (hunk 4 above)
Comment (midgleyc) [reply apps/editor/index.d.ts:585]: This one is actually a type bug fix. MarkdownEditor extends a CodeMirrorExt class:

https://github.com/nhn/tui.editor/blob/master/apps/editor/src/js/markdownEditor.js#L113

but this isn't represented in the types. I added two methods from CodeMirrorExt I found useful when developing custom commands: ideally I think they would all be added but this is quite a lot more work!

**Comment by midgleyc** [comment]
I have split this out into:

#986 which contains only the setValue function removal
#987 which includes some other type fixes and features I found essential while developing custom commands
#988 which contains the addEventType feature. As discussed in #970 and above this will wait until the specification is decided and may change.

Thanks for the review!
